lib/possible_bipartition.rb

```diff
@@ -1,4 +1,48 @@
+# Examples Input & Output
+# Input: dislikes = [ [],
+#                     [2, 3],
+#                     [1, 4],
+#                     [1],
+#                     [2]
+#                   ]
+# Output: true
+
+# If the first dog goes to group 1, any dog disliked by this dog must go to the opposite group -1. 
+# Then, any dog disliked by -1 dog is 1 and so on.
+# If at any point there is a conflict, the task is impossible.
+
+# BFS (using queue)
+# Time Complexity: O(n + e) - check every node and every edge
+# Space complexity: O(n) - using a queue
```

Comment:
Correct, except that your queue.shift probably costs you and makes it O(n^2 + e)